airbyte-webapp/src/views/Connector/ServiceForm/components/AuthButtonFactory.tsx

```diff
@@ -0,0 +1,58 @@
+import { Button } from "../../../../components/base/Button";
+import { FormattedMessage } from "react-intl";
+import { ConnectorDefinition } from "../../../../core/domain/connector";
+import { isSourceDefinition } from "../../../../core/domain/connector/source";
+import { isDestinationDefinition } from "../../../../core/domain/connector/destination";
+
+const getButton = (
+  loading: boolean,
+  run: () => void,
+  selectedService?: ConnectorDefinition,
+  done?: boolean
+): JSX.Element => {
+  let definitionId = "";
+  if (selectedService && isSourceDefinition(selectedService)) {
+    definitionId = selectedService.sourceDefinitionId;
+  } else if (selectedService && isDestinationDefinition(selectedService)) {
+    definitionId = selectedService.destinationDefinitionId;
+  } else {
+    throw new Error(`Unrecognized service type. Service: ${selectedService}`);
+  }
+
+  switch (definitionId) {
+    case "253487c0-2246-43ba-a21f-5116b20a2c50": // google ads
+    case "eff3616a-f9c3-11eb-9a03-0242ac130003": // google analytics
+    case "d19ae824-e289-4b14-995a-0632eb46d246": // google directory
+    case "eb4c9e00-db83-4d63-a386-39cfa91012a8": // google search console
+    case "71607ba1-c0ac-4799-8049-7f4b90dd50f7": // google sheets
+    case "ed9dfefa-1bbc-419d-8c5e-4d78f0ef6734": // google workspace admin reports
+      return (
+        <Button isLoading={loading} type="button" onClick={run} iconOnly>
+          {
+            <img
+              src="/connectors/google/btn_google_signin_dark_normal_web@2x.png"
+              height={40}
+              alt={"Sign in with google"}
```

Comment:
We use i18n for all messages, so it should use formatMessage

Reply:
@Jamakase this is almost certainly me not knowing front end very well - the Google image contains the "authenticate with google" text as well as the logo. Are you saying I should instead use the logo only next to our text? In this case do you mind giving a quick code example of what that would look like? Do I still add an img tag inside the button? How would I style the button (text/background/font etc) in a custom way while using <Button />? Any quick code sample would be helpful

Reply:
I just meant that the alt text you provided doesn't support i18n. We use react-intl lib fo rit. So you need to add this text into en.json and then use id as `formatMessage("some.id")` instead of providing text explicitly